go/interpreter/lfvm/instructions.go

```diff
@@ -167,7 +167,7 @@ func opMstore(c *context) {
 		c.status = statusError
 		return
 	}
-	if err := c.memory.SetWord(offset, value, c); err != nil {
+	if c.memory.SetWord(offset, value, c) != nil {
```

Comment:
Something to discuss: I kinda prefer the pattern
```
if err := call(..); err != nil {
```
over
```
if call(..) != nil {
```
since it indicates to the reader that what is returned is an error. Even if the `err` is not used below.
If you do not name the result, it could be anything that is of a reference type; a pointer, am map, a list, an interface, ...

Reply:
my motivation to delete the `err` in these kind of checks was that if we declare an error and compare it, it gives me the impression it should be propagated, but in these checks, we always do `if err := fun();  err != nil { c.signalErr(); return }` and that seemed very odd in my eyes. But I do understand that then one has to check what are all these functions returning. Either way seems like a compromise.

Reply:
In my work [here](https://github.com/Fantom-foundation/Tosca/compare/main...luis/lfvm_interpreter_test) I replaced everything back to `err := expresion; err != nil`. The reason being that errors continue propagating, and the if purpose is only to break execution.

IMO: an ignored error is a code smell, if we do not use errors, we could just return a boolean.